moveit_ros/planning/planning_scene_monitor/test/current_state_monitor_tests.cpp

```diff
@@ -0,0 +1,106 @@
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Tyler Weaver */
+
+#include <memory>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "moveit/planning_scene_monitor/current_state_monitor.h"
+#include "moveit/utils/robot_model_test_utils.h"
+#include "rclcpp/node_interfaces/node_clock_interface.hpp"
+#include "rclcpp/node_interfaces/node_topics_interface.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "tf2_ros/buffer.h"
+
+struct MockRclInterface : public planning_scene_monitor::CurrentStateMonitor::RclInterface
+{
+  MOCK_METHOD(rclcpp::Time, now, (), (const, override));
+  MOCK_METHOD(void, createJointStateSubscription,
+              (const std::string& topic, planning_scene_monitor::JointStateUpdateCallback callback), (override));
+  MOCK_METHOD(void, resetJointStateSubscription, (), (override));
+  MOCK_METHOD(std::string, getJointStateTopicName, (), (const, override));
+};
+
+TEST(StartStopTest, CurrentStateMonitorTests)
+{
+  auto mock_rcl_interface = std::make_unique<MockRclInterface>();
+  auto clock = std::make_shared<rclcpp::Clock>();
+  moveit::core::RobotModelPtr robot_model = moveit::core::loadTestingRobotModel("panda");
+  auto tf_buffer = std::make_shared<tf2_ros::Buffer>(clock);
+
+  EXPECT_CALL(*mock_rcl_interface, createJointStateSubscription);
+  EXPECT_CALL(*mock_rcl_interface, resetJointStateSubscription);
+  EXPECT_CALL(*mock_rcl_interface, now);
+
+  planning_scene_monitor::CurrentStateMonitor current_state_monitor(std::move(mock_rcl_interface), robot_model,
```

Comment:
Along the lines of scope limit
```
auto mock_rcl_interface = std::make_unique<MockRclInterface>{};
EXPECT_CALL(*mock_rcl_interface, createJointStateSubscription);
EXPECT_CALL(*mock_rcl_interface, resetJointStateSubscription);
EXPECT_CALL(*mock_rcl_interface, now);

planning_scene_monitor::CurrentStateMonitor current_state_monitor{
    std::move{mock_rcl_interface},
    moveit::core::loadTestingRobotModel{"panda"},
    std::make_shared<tf2_ros::Buffer>{std::make_shared<rclcpp::Clock>{}}};
```
might be a way to go, unless you think it's too noisy

Reply:
I like this.  I changed to doing this.  You can't use the {} initialization with std::move (it is a function) or std::make_shared (same reason I think).